### ISSUE TITLE
GetPackageDescription task caches description table without considering timestamps

### ISSUE THREAD

**Issue by safern** [dotnet/arcade#5445]
GetPackageDescription task caches the description table on the current MSBuild node using the full path to descriptions.json as the key. So if I'm adding a new package, do a full build of dotnet/runtime for example, get the failure that I need to add a description to the package. I go add it, then do a full build again, it will keep failing until I kill MSBuild nodes and try again.

Similar to: https://github.com/dotnet/arcade/commit/8820dcbdf6c93291b161a255c04f60b6b906c291#diff-38acd30f6f6e533a1d7b16bdb0ac3809

cc: @ericstj @Anipik @ViktorHofer 

FYI: @bartonjs who reached out as he was hitting this.

**Comment by bartonjs** [comment]
Let's just say it's really confusing when you build, then do a cherry pick or switch branches, and build again... and your build fails because it remembers what your hard drive used to look like.

**Comment by ericstj** [comment]
Yeah we should fix this; or just kill the whole descriptions.json and put this info in properties in the project files.  I'm not sure the value is there having this in a separate file.  Back when we started this pre-1.x there was an idea that we'd eventually localize all this but I don't think NuGet ever made the feature to do so.  Given the .NET SDK just uses properties in project files I'd prefer we do that and remove this thing.
